localstack/services/cloudformation/provider.py

```diff
@@ -153,6 +159,37 @@ def create_stack(self, context: RequestContext, request: CreateStackInput) -> Cr
                 )
             state.stacks.pop(existing.stack_id)
 
+        if (
+            "CAPABILITY_AUTO_EXPAND" not in request.get("Capabilities", [])
+            and "Transform" in template
+        ):
+            raise InsufficientCapabilitiesException(
+                "Requires capabilities : [CAPABILITY_AUTO_EXPAND]"
+            )
+
+        try:
+            parameters = template_preparer.resolve_parameters(
+                template.get("Parameters", {}), request.get("Parameters", [])
+            )
```

Comment:
nit: would move the template_preparer.resolve_parameters out of the try/except block.